src/main/java/co/zecko/retailer/client/httpClient/CartClient.java

```diff
@@ -0,0 +1,120 @@
+        }
+
+        queryParams.put("customerId",clientCustomerId);
+
+        String url = getUrl(BASE_URI + routeUri);
+
+        return get(url, queryParams, headers, new TypeReference<>() {});
+    }
+    public CartData addToCart(CartActionRequest cartActionRequest)
+        throws BaseException, IOException, InterruptedException{
+        zeckoAccessToken = getZeckoAccessToken(zeckoAccessToken);
+        Map<String, String> headers = getBaseHeaders(zeckoAccessToken);
+        String routeUri = "/add";
+        String url = getUrl(BASE_URI + routeUri);
+
+        return patch(url, cartActionRequest, headers, new TypeReference<>() {});
+    }
+
+    public CartData deleteFromCart(CartActionRequest cartActionRequest)
+            throws BaseException, IOException, InterruptedException{
+        zeckoAccessToken = getZeckoAccessToken(zeckoAccessToken);
+        Map<String, String> headers = getBaseHeaders(zeckoAccessToken);
+        String routeUri = "/delete";
+        String url = getUrl(BASE_URI + routeUri);
+
+        return patch(url, cartActionRequest, headers, new TypeReference<>() {});
+    }
+
+
+    public CartData update(String legacyDraftOrderId, CartUpdateRequest cartUpdateRequest)
+            throws BaseException, IOException, InterruptedException{
+        zeckoAccessToken = getZeckoAccessToken(zeckoAccessToken);
+        Map<String, String> headers = getBaseHeaders(zeckoAccessToken);
+        String routeUri = String.format("/%s/update", legacyDraftOrderId);
+        String url = getUrl(BASE_URI + routeUri);
+
+        return patch(url, cartUpdateRequest, headers, new TypeReference<>() {});
+    }
+
+    public CartData addDiscount(String legacyDraftOrderId, CartDiscountRequest cartDiscountRequest)
+            throws BaseException, IOException, InterruptedException{
+        zeckoAccessToken = getZeckoAccessToken(zeckoAccessToken);
+        Map<String, String> headers = getBaseHeaders(zeckoAccessToken);
+        String routeUri = String.format("/%s/add-discount", legacyDraftOrderId);
+        String url = getUrl(BASE_URI + routeUri);
+
+        return patch(url, cartDiscountRequest, headers, new TypeReference<>() {});
+    }
+
+    public CartData removeDiscount(String legacyDraftOrderId, CartDiscountRequest cartDiscountRequest)
+            throws BaseException, IOException, InterruptedException{
+        zeckoAccessToken = getZeckoAccessToken(zeckoAccessToken);
+        Map<String, String> headers = getBaseHeaders(zeckoAccessToken);
+        String routeUri = String.format("/%s/remove-discount", legacyDraftOrderId);
+        String url = getUrl(BASE_URI + routeUri);
+
+        return patch(url, cartDiscountRequest, headers, new TypeReference<>() {});
+    }
+
+    public OrderData complete(String legacyDraftOrderId, CartCompleteRequest cartCompleteRequest)
+            throws BaseException, IOException, InterruptedException{
+        zeckoAccessToken = getZeckoAccessToken(zeckoAccessToken);
```

Comment:
Add a check for presence of `legacyDraftOrderId` and `cartCompleteRequest`

Reply:
done